File: src/io/github/apollozhu/lottery/lottery/LotteryCenterPanelManagerPanel.java
package io.github.apollozhu.lottery.lottery;

import io.github.apollozhu.lottery.prize.LotteryPrizeDisplayPanel;
import io.github.apollozhu.lottery.prize.LotteryPrizeModel;

import javax.swing.*;
import java.awt.*;

public class LotteryCenterPanelManagerPanel extends JPanel {
    private CardLayout cardLayout = new CardLayout();

    private LotteryCenterPanel placeholder = new LotteryCenterPanel();
    private LotteryRollingPanel roller = new LotteryRollingPanel();
    private LotteryPrizeDisplayPanel winnerDisplay = new LotteryPrizeDisplayPanel();
    private LotteryCenterPanel center;

    public LotteryCenterPanelManagerPanel() {
        setLayout(cardLayout);
        add(placeholder, "placeholder");
        add(roller, "roller");
        add(winnerDisplay, "winnerDisplay");
    }

    public void showPlaceHodler() {
        cardLayout.show(this, "placeholder");
    }

    public void showRoller() {
        cardLayout.show(this, "roller");
    }

    public void showFor(String winner, LotteryPrizeModel prizeModel) {
        winnerDisplay.displayFor(winner, prizeModel);
        cardLayout.show(this, "winnerDisplay");
    }
}

File: src/io/github/apollozhu/lottery/lottery/LotteryCenterPanelManagerPanel.kt
package io.github.apollozhu.lottery.lottery

import io.github.apollozhu.lottery.prize.LotteryPrizeDisplayTextOnlyPanel
import io.github.apollozhu.lottery.prize.LotteryPrizeDisplayWithImagePanel
import io.github.apollozhu.lottery.prize.LotteryPrizeModel
import java.awt.CardLayout
import java.awt.event.MouseListener
import javax.swing.JPanel

class LotteryCenterPanelManagerPanel(clickListener: MouseListener) : JPanel() {
    private val cardLayout = CardLayout()

    private val placeholder = LotteryCenterPanel()
    private val roller = LotteryRollingPanel()
    private val displayWithImage = LotteryPrizeDisplayWithImagePanel()
    private val displayTextOnly = LotteryPrizeDisplayTextOnlyPanel()

    init {
        layout = cardLayout
        add(placeholder, "placeholder")
        add(roller, "roller")
        add(displayWithImage, "displayWithImage")
        add(displayTextOnly, "displayTextOnly")
        addMouseListener(clickListener)
    }

    fun showPlaceHodler() = cardLayout.show(this, "placeholder")

    fun showRoller() = cardLayout.show(this, "roller")

    fun showFor(winner: String, prizeModel: LotteryPrizeModel) {
        if (prizeModel.hasImage) {
            displayWithImage.displayFor(winner, prizeModel)
            cardLayout.show(this, "displayWithImage")
        } else {
            displayTextOnly.displayFor(winner, prizeModel)
            cardLayout.show(this, "displayTextOnly")
        }
    }
}
